composer/core/logging/logger_destination.py

```diff
@@ -63,3 +64,40 @@ def log_data(self, state: State, log_level: LogLevel, data: LoggerDataDict):
         """
         del state, log_level, data  # unused
         pass
+
+    def log_file_artifact(
+        self,
+        state: State,
+        log_level: LogLevel,
+        artifact_name: str,
+        file_path: pathlib.Path,
+        *,
+        allow_overwrite: bool,
+    ):
+        """Handle logging of a file artifact stored at ``file_path`` to an artifact named ``artifact_name``.
+
+        This method is invoked by the :class:`~composer.core.logging.logger.Logger`.
+
+        Implement this method to store the logged file.
```

Comment:
Maybe add a quick note that loggers don't strictly _need_ to implement this method? Some loggers just aren't appropriate for logging whole files.